packages/code-du-travail-frontend/src/modules/modeles-de-courriers/queries.ts

```diff
@@ -27,7 +29,44 @@ export const fetchAllModels = async <
     _source: fields,
     index: elasticDocumentsIndex,
   });
-  return response.hits.hits
-    .map(({ _source }) => _source)
-    .filter((source) => source !== undefined);
+  return response.hits.hits.map(({ _source }) => _source).filter(nonNullable);
+};
+
+export const fetchModel = async (
+  filter: Record<string, string>
+): Promise<
+  DocumentElasticResult<DocumentElasticWithSource<MailTemplateDoc>> | undefined
```

Comment:
```suggestion
  DocumentElasticResult<MailElasticDocument> | undefined
```

on peut utiliser le type `MailElasticDocument` dispo sur l'admin